### ISSUE TITLE
Error after adding library in Go project on Ubuntu 22.04 with Go 1.21

### ISSUE THREAD

**Issue by nikhil478** [bitcoin-sv/go-paymail#108]
**Description**:
After adding the `github.com/bitcoin-sv/go-paymail` library to my Go project, I encountered an error during the `go mod tidy` process. This issue occurs specifically on Ubuntu 22.04 with Go version 1.21, whereas the same setup works fine on Windows with Go version 1.20.x. Despite ensuring all my packages were updated, the error persists.

**Environment**:
- **OS**: Ubuntu 22.04
- **Go Version**: 1.21
- **Library Version**: github.com/bitcoin-sv/go-paymail (latest)

**Steps to Reproduce**:
1. Initialize a new Go project or use an existing one.
2. Add `github.com/bitcoin-sv/go-paymail` to the project's dependencies.
3. Run `go mod tidy`.

**Expected Behavior**:
`go build` should complete without any errors, updating the go.mod and go.sum files accordingly.

**Actual Behavior**:
An error occurs during the `go mod tidy` process.

**Error Output**:
```
go build -o neucron ./cmd/main.go
  chmod +x neucron
  shell: /usr/bin/bash -e {0}
  env:
    GOROOT: /opt/hostedtoolcache/go/1.21.11/x64
go: downloading github.com/sirupsen/logrus v1.9.0
go: downloading github.com/spf1[3](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:3)/viper v1.18.2
go: downloading github.com/getsentry/sentry-go v0.22.0
go: downloading github.com/lib/pq v1.10.7
go: downloading go.uber.org/zap v1.21.0
go: downloading github.com/fsnotify/fsnotify v1.7.0
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading github.com/sagikazarmark/slog-shim v0.1.0
go: downloading github.com/spf13/afero v1.11.0
go: downloading github.com/spf13/cast v1.6.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading golang.org/x/sys v0.21.0
go: downloading github.com/go-chi/chi/v5 v5.0.8
go: downloading github.com/google/uuid v1.6.0
go: downloading golang.org/x/crypto v0.2[4](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:4).0
go: downloading github.com/stripe/stripe-go/v7[5](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:5) v75.10.0
go: downloading github.com/stripe/stripe-go v70.15.0+incompatible
go: downloading github.com/golang-migrate/migrate/v4 v4.15.2
go: downloading github.com/jmoiron/sqlx v1.3.5
go: downloading gorm.io/driver/postgres v1.4.4
go: downloading gorm.io/gorm v1.24.0
go: downloading github.com/scylladb/gocql v1.10.0
go: downloading github.com/scylladb/gocqlx/v2 v2.8.0
go: downloading github.com/aws/aws-sdk-go v1.44.280
go: downloading github.com/go-playground/validator/v10 v10.20.0
go: downloading github.com/googollee/go-socket.io v1.7.0
go: downloading github.com/stripe/stripe-go/v7[6](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:6) v76.12.0
go: downloading go.opentelemetry.io/otel v1.24.0
go: downloading github.com/bitcoin-sv/go-paymail v0.19.0
go: downloading github.com/go-chi/cors v1.2.1
go: downloading github.com/go-chi/httprate v0.[7](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:7).4
go: downloading github.com/prometheus/client_golang v1.16.0
go: downloading github.com/swaggo/http-swagger v1.3.3
go: downloading gopkg.in/natefinch/lumberjack.v2 v2.2.1
go: downloading github.com/getsentry/sentry-go/otel v0.22.0
go: downloading go.opentelemetry.io/otel/sdk v1.24.0
go: downloading go.opentelemetry.io/otel/trace v1.24.0
go: downloading go.uber.org/atomic v1.11.0
go: downloading go.uber.org/multierr v1.11.0
go: downloading golang.org/x/text v0.16.0
go: downloading github.com/subosito/gotenv v1.6.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading github.com/magiconair/properties v1.[8](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:8).7
go: downloading github.com/pelletier/go-toml/v2 v2.2.2
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/go-resty/resty/v2 v2.13.1
go: downloading github.com/gofrs/uuid v4.4.0+incompatible
go: downloading github.com/libsv/go-bt v1.0.8
go: downloading github.com/jackc/pgx/v4 v4.18.1
go: downloading github.com/golang/snappy v0.0.4
go: downloading github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce2[9](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:9)ed
go: downloading github.com/klauspost/cpuid/v2 v2.2.7
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/scylladb/go-reflectx v1.0.1
go: downloading github.com/gomodule/redigo v1.8.4
go: downloading github.com/gabriel-vasile/mimetype v1.4.3
go: downloading github.com/go-playground/universal-translator v0.18.1
go: downloading github.com/leodido/go-urn v1.4.0
go: downloading golang.org/x/oauth2 v0.15.0
go: downloading github.com/mrz1836/go-api-router v0.5.3
go: downloading github.com/tonicpow/go-paymail v0.9.5
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/jackc/pgconn v1.14.0
go: downloading github.com/libsv/go-bk v0.1.6
go: downloading github.com/bitcoinschema/go-bitcoin v0.3.20
go: downloading github.com/libsv/go-bt/v2 v2.2.5
go: downloading golang.org/x/exp v0.0.0-20240409090435-93d18d7e34b8
go: downloading github.com/mrz1836/go-validate v0.2.0
go: downloading github.com/go-logr/logr v1.4.1
go: downloading go.opentelemetry.io/otel/metric v1.24.0
go: downloading github.com/bitcoinschema/go-bitcoin/v2 v2.0.5
go: downloading github.com/miekg/dns v1.1.61
go: downloading golang.org/x/net v0.26.0
go: downloading github.com/prometheus/client_model v0.3.0
go: downloading github.com/prometheus/common v0.42.0
go: downloading github.com/cespare/xxhash/v2 v2.2.0
go: downloading github.com/gorilla/websocket v1.5.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/prometheus/procfs v0.[10](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:10).1
go: downloading google.golang.org/protobuf v1.34.1
go: downloading github.com/swaggo/files v1.0.1
go: downloading github.com/swaggo/swag v1.16.3
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/jackc/pgio v1.0.0
go: downloading github.com/jackc/pgproto3/v2 v2.3.2
go: downloading github.com/jackc/pgtype v1.14.0
go: downloading github.com/jinzhu/inflection v1.0.0
go: downloading github.com/jinzhu/now v1.1.5
go: downloading github.com/bitcoinsv/bsvd v0.0.0-20190609155523-4c29707f7173
go: downloading github.com/bitcoinsv/bsvutil v0.0.0-20181216182056-1d77cf353ea9
go: downloading github.com/go-playground/locales v0.14.1
go: downloading cloud.google.com/go/compute/metadata v0.2.3
go: downloading cloud.google.com/go/compute v1.23.3
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading github.com/julienschmidt/httprouter v1.3.0
go: downloading github.com/matryer/respond v1.0.1
go: downloading github.com/mrz1836/go-logger v0.3.2
go: downloading github.com/mrz1836/go-parameters v0.3.1
go: downloading github.com/newrelic/go-agent/v3/integrations/nrhttprouter v1.0.2
go: downloading github.com/newrelic/go-agent/v3 v3.26.0
go: downloading github.com/mrz1836/go-sanitize v1.3.1
go: downloading github.com/jackc/chunkreader/v2 v2.0.1
go: downloading github.com/jackc/pgpassfile v1.0.0
go: downloading github.com/jackc/pgservicefile v0.0.0-20221227161230-091c0ba34f0a
go: downloading github.com/golang-jwt/jwt/v4 v4.5.0
go: downloading github.com/lestrrat-go/jwx/v2 v2.0.6
go: downloading github.com/go-logr/stdr v1.2.2
go: downloading github.com/libsv/go-bc v0.1.29
go: downloading github.com/golang/protobuf v1.5.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.4
go: downloading github.com/KyleBanks/depth v1.2.1
go: downloading github.com/go-openapi/spec v0.21.0
go: downloading golang.org/x/tools v0.22.0
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/ugorji/go/codec v1.2.12
go: downloading github.com/valyala/fasthttp v1.50.0
go: downloading google.golang.org/grpc v1.59.0
go: downloading github.com/lestrrat-go/iter v1.0.2
go: downloading github.com/lestrrat-go/option v1.0.0
go: downloading github.com/bitcoinsv/bsvlog v0.0.0-20181216181007-cb81b076bf2e
go: downloading github.com/go-openapi/jsonpointer v0.21.0
go: downloading github.com/go-openapi/jsonreference v0.21.0
go: downloading github.com/go-openapi/swag v0.23.0
go: downloading github.com/libsv/go-p2p v0.1.5
go: downloading github.com/andybalholm/brotli v1.0.5
go: downloading github.com/klauspost/compress v1.17.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-2023[11](https://github.com/timechainlabs/neucron-server/actions/runs/9829908821/job/27135492682#step:4:12)20223509-83a465c0220f
go: downloading github.com/lestrrat-go/blackmagic v1.0.1
go: downloading github.com/lestrrat-go/httprc v1.0.4
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/lestrrat-go/httpcc v1.0.1
go: downloading github.com/josharian/intern v1.0.0
go: updates to go.mod needed; to update it:
	go mod tidy
Error: Process completed with exit code 1.
```

**Additional Information**:
- This issue is specific to Ubuntu 22.04 with Go 1.21.
- The project works fine with Go 1.20.x on Windows.
- All packages are updated to their latest versions.

**Comment by nikhil478** [comment]
@sirdeggen @proapi @dorzepowski

**Comment by nikhil478** [comment]
changing indirect dependency version of these library 
github.com/klauspost/cpuid/v2 v2.2.7 // indirect
	github.com/leodido/go-urn v1.4.0 // indirect
	github.com/libsv/go-p2p v0.1.5
to 
github.com/klauspost/cpuid/v2 v2.2.5 // indirect
	github.com/leodido/go-urn v1.2.4 // indirect
	github.com/libsv/go-p2p v0.1.3 // indirect

resolve my issue

**Comment by dorzepowski** [comment]
**Reproduction of the Issue:**
I tried to reproduce the issue.
1. I used ubuntu 22.04
2. I installed go `go1.21.12` on it
3. I created new directory and go module `example`
4. I copied [get_capabilities.go](https://raw.githubusercontent.com/bitcoin-sv/go-paymail/main/examples/client/get_capabilities/get_capabilities.go) from go-paymail examples as main.go in this new project
5. I used `go get github.com/bitcoin-sv/go-paymail`
6. then I used `go mod tidy`
7. and then `go build`

**Result:**
go build passed without the issue and the binary was runnable.


**Summary:**
I'm closing this issue because I cannot reproduce this issue and it seems to be rather related to your project and it's dependencies and go dependencies management.
